### PR TITLE
fix(code): fix some lgtm analysis warnings

### PR DESCRIPTION
**Pull request by philbooth**
Ref: https://lgtm.com/projects/g/mozilla/fxa-content-server/alerts/?mode=list

@mozilla/fxa-devs r?

### CODE REVIEW

File: server/lib/ga-collector.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ GACollector.prototype = {
    * @param {Object} body
    */
   write: function (body) {
-    if (! body || ! body.events || ! body.events.length > 0 || ! this.analyticsId) {
+    if (! body || ! body.events || body.events.length === 0 || ! this.analyticsId) {
```
Comment (philbooth) [inline server/lib/ga-collector.js]: In classic JS fashion, I think this one was actually behaving as intended before, even though it made an incorrect assumption about the relative precedence of the `!` and `>` operators:

* If `length` is greater than zero, `! length` is false, which coerces to zero, which doesn't enter the condition.

* If `length` is zero, `! length` is true`, which coerces to one, which does enter the condition.

Good old JS, it's always got your back.

File: app/scripts/views/account_recovery_confirm_key.js [hunk 2]
```diff
@@ -32,7 +32,6 @@ const View = FormView.extend({
   },
 
   initialize(options) {
```
Comment (vbudhram) [inline app/scripts/views/account_recovery_confirm_key.js]: You could probably remove from function definition too.

File: app/scripts/views/settings/avatar_crop.js [hunk 3]
```diff
@@ -26,8 +26,6 @@ define(function (require, exports, module) {
     viewName: 'settings.avatar.crop',
 
     initialize (options) {
```
Comment (vbudhram) [inline app/scripts/views/settings/avatar_crop.js]: Same as above, can remove from function.

File: app/scripts/lib/mailcheck.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ define(function (require, exports, module) {
   const _ = require('underscore');
   const $ = require('jquery');
   const KeyCodes = require('./key-codes');
-  const mailcheck = require('mailcheck'); //eslint-disable-line no-unused-vars
+  require('mailcheck');
```
[the remark this answers is not in the document]
Comment (philbooth) [reply app/scripts/lib/mailcheck.js]: Don't know for certain but I presume it's some kind of programming-by-side-effects kind of deal.

**Pull request merged**